.github/workflows/pre-commit.yaml

```diff
@@ -10,24 +9,11 @@ jobs:
   pre-commit:
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout
+      - name: Checkout main code
         uses: actions/checkout@v2
 
-      - name: Set up Python
-        uses: actions/setup-python@v2
+      - name: Pre-Commit Action
+        uses: ccao-data/actions/pre-commit@create-composite-pre-commit-action
         with:
-          python-version: 3.10.12
-
-      - name: Install pre-commit
-        run: sudo apt-get install pre-commit
-        shell: bash
-
-      - name: Cache pre-commit environment
-        uses: actions/cache@v2
-        with:
-          path: ~/.cache/pre-commit
-          key: pre-commit-${{ hashFiles('.pre-commit-config.yaml') }}
-
-      - name: Run pre-commit
-        run: pre-commit run --show-diff-on-failure --color=always --all-files
-        shell: bash
+          hooks: "shellcheck yamllint black"
+          python-version: '3.10.12'
```

Comment:
**[Nitpick, non-blocking]** I think this isn't necessary since it's not used by [the action](https://github.com/ccao-data/actions/pull/22/files#diff-5769c037baef7e92c33706e693ad9cdb4ae31f2632659aa1ec0f0544979b8f0e) anymore, right?

```suggestion
```